scipy/signal/spectral.py

```diff
@@ -141,7 +141,7 @@ def periodogram(x, fs=1.0, window=None, nfft=None, detrend='constant',
                  scaling, axis)
 
 
-def welch(x, fs=1.0, window='hann', nperseg=256, noverlap=None, nfft=None,
+def welch(x, fs=1.0, window='hann', nperseg=None, noverlap=None, nfft=None,
```

Comment:
@e-q all functions still default to 256 where stated in the docstrings. It happens inside ```_triage_segments```.

I.e., as I have modified it, when someone calls any of these functions (except periodogram) one of three things happens:
1. they supply a string for window, e.g., 'hann', and then if nperseg==None, it is set to 256
2. they supply an ndarray, and if nperseg==None, it is set to the length of the ndarray
3. they supply an ndarray, and also a value for nperseg; if nperseg==len(ndarray), the expected happens, and if nperseg != len(ndarray), an error is thrown

That's the logic I came up with to resolve the original issue. I realize it's confusing that the docstring says one thing but the defaults in the keyword list say another.